scheduler/cloudformation/cloudformation.go

```diff
@@ -911,26 +944,144 @@ func (m *Scheduler) Run(ctx context.Context, app *scheduler.App, process *schedu
+		}
+		// NOTE: Currently, this depends on a patched version of the
+		// Amazon ECS Container Agent, since the official agent doesn't
+		// provide a method to pass these down to the `CreateContainer`
+		// call.
+		containerDefinition.DockerLabels["docker.config.Tty"] = aws.String("true")
+		containerDefinition.DockerLabels["docker.config.OpenStdin"] = aws.String("true")
+	}
+
 	resp, err := m.ecs.RegisterTaskDefinition(&ecs.RegisterTaskDefinitionInput{
 		Family: aws.String(fmt.Sprintf("%s--%s", app.ID, process.Type)),
 		ContainerDefinitions: []*ecs.ContainerDefinition{
-			t.ContainerDefinition(app, process),
+			containerDefinition,
 		},
 	})
 	if err != nil {
 		return fmt.Errorf("error registering TaskDefinition: %v", err)
 	}
 
-	_, err = m.ecs.RunTask(&ecs.RunTaskInput{
-		TaskDefinition: resp.TaskDefinition.TaskDefinitionArn,
-		Cluster:        aws.String(m.Cluster),
-		Count:          aws.Int64(1),
-		StartedBy:      aws.String(app.ID),
+	var constraint []*ecs.PlacementConstraint
+	if m.PlacementConstraints != nil {
+		if attached {
+			constraint = m.PlacementConstraints.Attached
+		} else {
+			constraint = m.PlacementConstraints.Detached
+		}
+	}
+
+	runResp, err := m.ecs.RunTask(&ecs.RunTaskInput{
+		TaskDefinition:       resp.TaskDefinition.TaskDefinitionArn,
+		Cluster:              aws.String(m.Cluster),
+		Count:                aws.Int64(1),
+		StartedBy:            aws.String(app.ID),
+		PlacementConstraints: constraint,
 	})
 	if err != nil {
 		return fmt.Errorf("error calling RunTask: %v", err)
 	}
 
+	for _, f := range runResp.Failures {
+		return fmt.Errorf("error running task %s: %s", aws.StringValue(f.Arn), aws.StringValue(f.Reason))
+	}
+
+	task := runResp.Tasks[0]
+
+	if attached {
+		if err := m.attach(ctx, task, in, out); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// attach attaches to the given ECS task.
+func (m *Scheduler) attach(ctx context.Context, task *ecs.Task, in io.Reader, out io.Writer) error {
+	defer tryClose(out)
+
+	descContainerInstanceResp, err := m.ecs.DescribeContainerInstances(&ecs.DescribeContainerInstancesInput{
+		Cluster:            task.ClusterArn,
+		ContainerInstances: []*string{task.ContainerInstanceArn},
+	})
+	if err != nil {
+		return fmt.Errorf("error describing container instance (%s): %v", aws.StringValue(task.ContainerInstanceArn), err)
+	}
+
+	containerInstance := descContainerInstanceResp.ContainerInstances[0]
+	descInstanceResp, err := m.ec2.DescribeInstances(&ec2.DescribeInstancesInput{
+		InstanceIds: []*string{containerInstance.Ec2InstanceId},
+	})
+	if err != nil {
+		return fmt.Errorf("error describing ec2 instance (%s): %v", aws.StringValue(containerInstance.Ec2InstanceId), err)
+	}
+
+	ec2Instance := descInstanceResp.Reservations[0].Instances[0]
+
+	// Wait for the task to start running. It will stay in the
+	// PENDING state while the container is being pulled.
+	if err := m.ecs.WaitUntilTasksRunning(&ecs.DescribeTasksInput{
```

Comment:
This should actually wait until it's RUNNING or STOPPED. If you run something that returns quickly (e.g. `emp run echo hello world`), the task could hit `STOPPED` before this method had a change to actually see it was running. It should be fine to `AttachToContainer` on a stopped task, since the ECS agent waits 3 hours to cleanup containers.